driver-core/src/main/java/com/datastax/driver/core/exceptions/ConnectionException.java

```diff
@@ -13,29 +13,43 @@
  *   See the License for the specific language governing permissions and
  *   limitations under the License.
  */
-package com.datastax.driver.core;
+package com.datastax.driver.core.exceptions;
 
+import java.net.InetAddress;
 import java.net.InetSocketAddress;
 
-class ConnectionException extends Exception {
+public class ConnectionException extends DriverException implements CoordinatorException {
```

Comment:
Had to make this exception and its subclasses public because they can now be exposed to user code.